docs/modules/serialization/pages/serialization.adoc

```diff
@@ -1,32 +1,75 @@
-stored in different partitions on multiple cluster members.
-The data you need may not be present on the local member, and in that case,
-Hazelcast retrieves that data from another member. This requires serialization.
+{description}
 
-Serialization is used in the following cases:
+== Serialization in Client/Server Mode
 
-* Adding key/value objects to a map
-* Putting items in a queue/set/list
-* Sending a runnable using an executor service
-* Processing an entry within a map
-* Locking an object
-* Sending a message to a topic
+In xref:overview:topology.adoc[client/server mode], you can serialize your data on the client side and/or the xref:ROOT:glossary.adoc#member[member] (server) side.
 
-Hazelcast optimizes the serialization for the basic types and their array types
-with its built-in serializers. By default, you cannot override this behavior; however,
-Hazelcast offers a way to override them using a configuration option,
-see the xref:overriding-built-in-serializers.adoc[Overriding the Default Serializers section].
-
-Hazelcast's built-in serializers can handle the following types:
+If you just want to store data on members, clients can handle serialization without the members needing to know how to do so. For example, you can serialize data as JSON and add it to a map. Then you can later request that information and deserialize on the client.
+
+However, in the following cases, you must configure members to be able to serialize and deserialize your data:
+
+[cols="a,a,a,a"]
+|===
+|Case|Clients|Reason|Exceptions
+
+|When the data structure's values are xref:data-structures:map.adoc#setting-in-memory-format[stored as objects]
+|All clients
+|Members must be able to serialize or deserialize the object in order to send it to other members or receive it from other members
+|
+
+|When you want to xref:computing:entry-processor.adoc[process map entries] or use an xref:computing:executor-service.adoc[executor service]
+|All clients
+|Members need to serialize the tasks
+|When using the Java client with xref:implementing-portable-serialization.adoc[`Portable`] serialization, you don't need to configure serialization on the member. For details, see xref:clusters:accessing-domain-objects.adoc[].
+
+|When you use xref:sql:querying-imap.adoc#key-and-value-fields[SQL queries on nested fields] whose values are not primitives
+|Java only
+|Members need to deserialize underlying objects to be able to read nested fields
+|You can query nested fields whose values are primitives because these data types are serialized by Hazelcast by default. See <<data-types-with-built-in-serialization, Data Types with Built-In Serialization>>
+|===
+
+=== Serialization Options for Clients
+
+The following serialization options are available for clients:
+
+- IdentifiedDataSerializable
+- Portable
+- Custom Serialization
+
+For a comparison of these options, see xref:comparing-interfaces.adoc[].
+
+== Serialization in Embedded Mode

```

Comment:
Not really sure on that one but, I feel like this section is redundant. For anything that you do on hazelcast, the object needs to be serialized. So the list below is not really complete anyway.

Reply:
True. This list came from the existing docs. Maybe we can point out that these are examples:

In xref:overview:topology.adoc[embedded mode] data is partitioned across members in a cluster. As a result, each member must be able to serialize and deserialize all data in the cluster in order to send and receive it. For example:

-
-
-